CockpitPlugin/src/main/java/com/polidea/cockpitplugin/generator/DebugCockpitGenerator.kt

```diff
@@ -25,58 +26,138 @@ internal class DebugCockpitGenerator : BaseCockpitGenerator() {
+            is CockpitColor -> String::class.java
+            else -> mapToTypeClass(param)
+        }
+        val builder = MethodSpec.methodBuilder("set${param.name.capitalize()}")
                 .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
                 .addParameter(ParameterSpec.builder(typeClass, param.name).build())
-                .addStatement("\$T.INSTANCE.setParamValue(\"${param.name}\", ${param.name})",
-                        cockpitManagerClassName)
-                .build()
+        when (param.value) {
+            is CockpitColor -> {
+                builder.addStatement("\$T cockpitColor = cockpitColorMapper.wrap(color)", cockpitColorClassName)
+                builder.addStatement("\$T.INSTANCE.setParamValue(\"${param.name}\", cockpitColor)", cockpitManagerClassName)
+            }
+            else -> builder.addStatement("\$T.INSTANCE.setParamValue(\"${param.name}\", ${param.name})",
+                    cockpitManagerClassName)
+        }
+        return builder.build()
     }
 
     internal fun createAddPropertyChangeListenerMethodSpecForParam(param: CockpitParam<*>): MethodSpec {
-        val typeClass = mapToJavaObjectTypeClass(param)
+        val typeClass = when (param.value) {
+            is CockpitColor -> String::class.java
+            else -> mapToJavaObjectTypeClass(param)
+        }
         val listenerParamName = "listener"
-        return MethodSpec.methodBuilder("addOn${param.name.capitalize()}ChangeListener")
+        val builder = MethodSpec.methodBuilder("addOn${param.name.capitalize()}ChangeListener")
                 .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
                 .addParameter(ParameterSpec.builder(getParametrizedCockpitPropertyChangeListenerClassName(typeClass), listenerParamName).build())
-                .addStatement("\$T.INSTANCE.addOnParamChangeListener(\"${param.name}\", $listenerParamName)",
-                        cockpitManagerClassName)
+        when (param.value) {
+            is CockpitColor -> {
+                builder.addStatement("\$T colorListener = new \$T($listenerParamName, cockpitColorMapper)",
+                        getParametrizedCockpitPropertyChangeListenerClassName(cockpitColorClassName),
+                        ParameterizedTypeName.get(mappingPropertyChangeListenerClassName, cockpitColorClassName, TypeName.get(String::class.java)))
+                builder.addStatement("colorListenerMap.put($listenerParamName, colorListener)")
+                builder.addStatement("\$T.INSTANCE.addOnParamChangeListener(\"${param.name}\", colorListener)", cockpitManagerClassName)
+            }
+            else -> builder.addStatement("\$T.INSTANCE.addOnParamChangeListener(\"${param.name}\", $listenerParamName)",
+                    cockpitManagerClassName)
+        }
+        return builder
                 .build()
     }
 
     internal fun createRemovePropertyChangeListenerMethodSpecForParam(param: CockpitParam<*>): MethodSpec {
```

Comment:
Because we don't have simple types anymore and check the type in `generate` method, what about this:

`
private fun <T : Any> createRemovePropertyChangeListenerMethodSpecForParam(name: String, value: T): MethodSpec
`